homeassistant/components/tahoma/__init__.py

```diff
@@ -1,149 +1,201 @@
-"""Support for Tahoma devices."""
+"""The TaHoma integration."""
+import asyncio
 from collections import defaultdict
+from datetime import timedelta
 import logging
 
-from requests.exceptions import RequestException
-from tahoma_api import Action, TahomaApi
+from aiohttp import ClientError, ServerDisconnectedError
+from pyhoma.client import TahomaClient
+from pyhoma.exceptions import BadCredentialsException, TooManyRequestsException
+from pyhoma.models import Command
 import voluptuous as vol
 
+from homeassistant.components.scene import DOMAIN as SCENE
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_EXCLUDE, CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import config_validation as cv, discovery
-from homeassistant.helpers.entity import Entity
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+
+from .const import (
+    CONF_UPDATE_INTERVAL,
+    DEFAULT_UPDATE_INTERVAL,
+    DOMAIN,
+    IGNORED_TAHOMA_TYPES,
+    TAHOMA_TYPES,
+)
+from .coordinator import TahomaDataUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
-DOMAIN = "tahoma"
-
-TAHOMA_ID_FORMAT = "{}_{}"
+SERVICE_EXECUTE_COMMAND = "execute_command"
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_USERNAME): cv.string,
-                vol.Required(CONF_PASSWORD): cv.string,
-                vol.Optional(CONF_EXCLUDE, default=[]): vol.All(
-                    cv.ensure_list, [cv.string]
-                ),
-            }
+        DOMAIN: vol.All(
+            cv.deprecated(CONF_EXCLUDE),
```

Comment:
I am aware that HA no longer accepts YAML configuration for integrations that connect to devices or services. (ADR-0010: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision).

Is this change allowed? We are not using the `exclude` param anymore, since you can disable entities now via core functionality. It won't harm when you pass them, they will simply be ignored, thus I think that marking `CONF_EXCLUDE` as deprecated should be allowed.

Reply:
A deprecation is ok.